projects/kit/internal/primitive-calendar-range/primitive-calendar-range.component.ts

```diff
@@ -110,6 +110,7 @@ export class TuiPrimitiveCalendarRangeComponent implements OnInit {
 
     ngOnInit() {
         this.updateViewedMonths();
+        this.setInitialMonths();
```

Comment:
```ts
this.updateViewedMonths();
```
This old function set values to `this.userViewedMonthFirst` and `this.userViewedMonthSecond` too (as your new function `this.setInitialMonths();`).

This functions are both synchronous, so the second function overrides the second one.
Do we need both functions ?

Reply:
yes, `this.updateViewedMonths()` set values to `this.userViewedMonthFirst` and `this.userViewedMonthSecond` every value change. But `setInitialValues` we should call only once (onInit) to clamp `this.userViewedMonthFirst` and `this.userViewedMonthSecond` between max/min

Reply:
I'll add unit for this case 👍

Reply:
Then maybe we do not need `this.updateViewedMonths();` in `ngOnInit`?

Reply:
thanks, `this.updateViewedMonths();` deleted from `ngOnInit`